creator/refferalTokens/models.py

```diff
@@ -0,0 +1,49 @@
+import uuid
+from django.db import models
+from creator.studies.models import Study
+
+User = get_user_model()
+
+
+class ReferralToken(models.Model):
+    """
+    An Event that involved some user, study, file, version, or a combination
+    of them.
+    """
+
+    class Meta:
+        permissions = [
+            ("add_refferaltoken", "Can generate a new referral token",)
+        ]
+
+    id = models.UUIDField(default=uuid.uuid4, unique=True, editable=False)
+
+    email = models.CharField(
+        max_length=150,
+        unique=True,
+        help_text="The email that the token link will be sent to",
+    )
+
+    claimed = models.BooleanField(
+        default=False, help_text="If the token has been used"
+    )
+
+    study = models.ForeignKey(
+        Study,
+        related_name="studys",
+        help_text="List of studies that the user will be added to",
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+
+    role = models.CharField(
```

Comment:
This should called `groups` and be a many-many to `django.contrib.auth.models.Group`